draft-parecki-oauth-client-id-metadata-document.md

```diff
@@ -102,12 +117,37 @@ The client metadata document URL is a JSON document containing the metadata
 of the client. The client metadata values are the values defined in
 OAuth Dynamic Client Registration ({{RFC7591}}) section 2.
 
+The client metadata document MUST contain a `client_id` property whose value
+MUST compare and match the URL of the document using simple string comparison as
+defined in [RFC3986] Section 6.2.1.
+
+The client metadata document MAY define additional properties in the response.
+The client metadata document MAY also be served with more specific content types
+as long as the response is JSON and conforms to `application/<AS-defined>+json`.
+
+Other specifications MAY place additional restrictions on the contents of the
+client metadata document accepted by authorization servers implementing their
+specification, for instance, preventing the registration of confidential clients
+by requiring the `token_endpoint_auth_method` property be set to `"none"`.
 
 ## Metadata Discovery Errors
 
-If fetching the metadata document fails, the authorization server MAY abort the
-authorization request, or continue with the information it has available.
+If fetching the metadata document fails, the authorization server SHOULD abort the
+authorization request.
+
+## Metadata Caching
+
+The authorization server MAY cache the client metadata it discovers at the
+client metadata document URL.
 
+TBD: recommend a cache lifetime? considerations about stale data?
```

Comment:
Maybe the AS should respect the HTTP cache headers that the client returns from the URL?

Reply:
I was maybe thinking etags / if-unmodified-since type headers? I'm really not sure